ts/Extensions/Annotations/Popup.ts

```diff
@@ -865,143 +944,388 @@ H.Popup.prototype = {
+                filteredSeries: Highcharts.FilteredSeries;
+
+            objectEach(series, function (
+                series: Series,
+                value: string
+            ): void {
+                const seriesOptions = series.options;
+                // Allow only indicators.
+                if (
+                    (series as any).params || seriesOptions &&
+                    (seriesOptions as any).params
+                ) {
+                    const { indicatorFullName, indicatorType } =
+                        indicators.getNameType(series, value);
+
+                    if (filter) {
+                        // Replace invalid characters.
+                        const validFilter = filter.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+                        const regex = new RegExp(validFilter, 'i'),
+                            alias = indicatorAliases &&
+                                indicatorAliases[indicatorType] &&
+                                indicatorAliases[indicatorType].join(' ') || '';
+
+                        if (indicatorFullName.match(regex) || alias.match(regex)) {
+                            filteredSeries = {
+                                indicatorFullName,
+                                indicatorType,
+                                series
+                            };
+
+                            filteredSeriesArray.push(filteredSeries);
+                        }
+                    } else {
+                        filteredSeries = {
+                            indicatorFullName,
+                            indicatorType,
+                            series
+                        };
+
+                        filteredSeriesArray.push(filteredSeries);
+                    }
+                }
+            });
+
+            return filteredSeriesArray;
+        },
+
+        /**
+         * Filter an array of series and map its names and types.
+         *
+         * @private
+         *
+         * @param {Highcharts.FilteredSeries} series
+         *        All series that are available in the plotOptions.
+         */
+        filterSeriesArray: function (
+            this: Highcharts.Popup,
+            series: Array<Series>
+        ): [Highcharts.FilteredSeries] {
+            let filteredSeriesArray: [Highcharts.FilteredSeries] = [] as any,
+                filteredSeries: Highcharts.FilteredSeries;
+
+            // Allow only indicators.
+            series.forEach(function (series): void {
+                const seriesOptions = series.options;
+
+                if (series.is('sma')) {
+                    filteredSeries = {
+                        indicatorFullName: series.name,
+                        indicatorType: series.type,
+                        series
+                    };
+
+                    filteredSeriesArray.push(filteredSeries);
+                }
+            });
+
+            return filteredSeriesArray;
```

Comment:
The function doclet has no return information. The return type is a singlet with FilteredSeries. Though the loop can push more than one series. If it is really just one, then just return the first series directly out of the loop (classic for).

Reply:
And otherwise the return type should be a regular array (`Array<Highcharts.FilteredSeries>`).